File: src/Java/Behavioral/Strategy/DispatchStrategy/SMSStrategy.java
package Java.Behavioral.Strategy.DispatchStrategy;

import Java.Behavioral.Strategy.Message;

public class SMSStrategy implements DispatchStrategy {
    @Override
    public void send(Message message) {
        System.out.println("Message sent to " + message.getReceivingUserDetails().getFullName() + " via SMS");
        System.out.println("Content: " + message.getContent());
        System.out.println("Send Timestamp: " + message.getDateTimeSent() + "\n");
    }
}

File: src/Kotlin/strategyPattern/DispatchStrategies/SMSStrategy.kt
package Kotlin.strategyPattern.DispatchStrategies

import Kotlin.strategyPattern.Message

class SMSStrategy : DispatchStrategy {

    override fun send(message: Message) {
        println("Message sent to " + message.receivingUser.fullName + "Via SMS")
        println("Content: " + message.content)
        println("Send Timestamp: " + message.dateTimeSent + "\n")
    }
}